os/hal/ports/SN32/LLD/SN32F2xx/USB/hal_usb_lld.c

```diff
@@ -459,6 +460,24 @@ void handleACK(USBDriver* usbp, usbep_t ep) {
     }
 }
 
+void handleNAK(USBDriver *usbp, usbep_t ep) {
+    uint8_t out = 0;
+
+    if(ep > 0 && ep <= USB_MAX_ENDPOINTS) {
+        out = ( SN32_USB->CFG & mskEPn_DIR(ep) ) == mskEPn_DIR(ep);
+    }
+    else {
+        return;
+    }
+
+    if(out)
+    {
+        // no ack required here
+    }
+    else{
```

Comment:
this can probably be simplified now to something like
`if(!out) {`

Reply:
That what I head first, but then I thought it might be better to have the out case explicitly ignored for future person who may stumble upon this code